accounts/abi/bind/backends/simulated.go

```diff
@@ -60,8 +60,8 @@ func NewSimulatedBackend(accounts ...core.GenesisAccount) *SimulatedBackend {
 // Commit imports all the pending transactions as a single block and starts a
 // fresh new state.
 func (b *SimulatedBackend) Commit() {
-	if _, err := b.blockchain.InsertChain([]*types.Block{b.pendingBlock}); err != nil {
-		panic(err) // This cannot happen unless the simulator is wrong, fail in that case
+	if res := b.blockchain.InsertChain([]*types.Block{b.pendingBlock}); res.Error != nil {
```

Comment:
This is an example of a change where `core/` `*Insert` methods return a dedicated response type, instead of a `(n int, err error)` pattern. This allows more transparent and less redundant information sharing between downloader, blockchain, and eventing and logging contexts.